dask/dataframe/groupby.py

```diff
@@ -293,6 +293,95 @@ def _var_agg(g, levels, ddof):
     return result
 
 
+def _cov_combine(g, levels):
+    return g
+
+
+def _cov_finalizer(df, cols):
+    vals = []
+    num_elements = len(list(it.product(cols, repeat=2)))
+    num_cols = len(cols)
+    vals = list(range(num_elements))
+    col_idx_mapping = dict(zip(cols, range(len(cols))))
+    for i, j in it.combinations_with_replacement(df[cols].columns, 2):
+        x = col_idx_mapping[i]
+        y = col_idx_mapping[j]
+        idx = x + num_cols * y
+        mul_col = "%s%s" % (i, j)
+        ni = df["%s-count" % i]
+        nj = df["%s-count" % j]
+
+        n = np.sqrt(ni * nj)
+        div = n - 1
+        div[div < 0] = 0
+        val = (df[mul_col] - df[i] * df[j] / n).values[0] / div.values[0]
+
+        vals[idx] = val
+        if i != j:
+            idx = num_cols * x + y
+            vals[idx] = val
+
+    level_1 = cols
+    index = pd.MultiIndex.from_product([level_1, level_1])
+    return pd.Series(vals, index=index)
+
+
+def mul_cols(df, cols):
+    _df = type(df)()
+    for i, j in it.combinations_with_replacement(df[cols].columns, 2):
+        col = "%s%s" % (i, j)
+        _df[col] = df[i] * df[j]
+    return _df
+
+
+def _cov_chunk(df, *index):
+    if is_series_like(df):
+        df = df.to_frame()
+    df = df.copy()
+    cols = df._get_numeric_data().columns
+
+    # when grouping by boolean series don't drop columns
```

Comment:
I'm confused by this comment, as I don't see any boolean dtype checks.

Reply:
This happened in  `test_dataframe_aggregations_multilevel` when groupper is `['a'] > 2, df['b'] > 1]`.  Usually, I am dropping the column(s) which will become the index after the groupby.  When a boolean mask is used there is no need.  This is all done in service of getting a list of columns to be used by mul_cols -- perhaps there is a better way of getting a list of columns I need to multiply against one another

Reply:
Gotcha, so the issue isn't that it's a boolean mask, it's that it's an external Series? `df.groupby([series_1, series_2]).cov()` would have the same issue, regardless of whether those series are bool / int / etc?

> perhaps there is a better way of getting a list of columns I need to multiply against one another
 
I see this logic in a few places (e.g. GroupBy.aggregate). I don't currently see a good central place for it.

Reply:
I am going through this now but it's a little tricky becuase you may have to munge the index as well.  For example, we need to handle groupby(['a', 'b'])  where a->1, b->2, c->3